source/common/network/proxy_protocol.cc

```diff
@@ -66,11 +66,31 @@ void ProxyProtocol::ActiveConnection::onReadWorker() {
   // Remove the line feed at the end
   StringUtil::rtrim(proxy_line);
 
-  // Parse proxy protocol line with format: PROXY TCP4/TCP6 SOURCE_ADDRESS DESTINATION_ADDRESS
-  // SOURCE_PORT DESTINATION_PORT.
+  // Parse proxy protocol line with format: PROXY TCP4/TCP6/UNKNOWN SOURCE_ADDRESS
+  // DESTINATION_ADDRESS SOURCE_PORT DESTINATION_PORT.
   const auto line_parts = StringUtil::split(proxy_line, " ", true);
 
-  if (line_parts.size() != 6 || line_parts[0] != "PROXY") {
+  if (line_parts.size() < 2 || line_parts[0] != "PROXY") {
+    throw EnvoyException("failed to read proxy protocol");
+  }
+
+  if (line_parts[1] == "UNKNOWN") {
+    // At this point we know it's a proxy protocol line, so we can remove it from the socket
+    // and continue.
+    Address::InstanceConstSharedPtr local_address = Envoy::Network::Address::addressFromFd(fd_);
+    Address::InstanceConstSharedPtr remote_address;
+    // The remote address not known.
+    if (local_address->ip()->version() == Address::IpVersion::v4) {
+      remote_address = std::make_shared<Address::Ipv4Instance>(Address::Ipv4Instance("0.0.0.0"));
+    } else {
+      remote_address = std::make_shared<Address::Ipv6Instance>(Address::Ipv6Instance("::"));
```

Comment:
Is there a way to get coverage on this branch? I think it should be possible with the integration test? If it's super hard it's ok to skip but would be nice to have. (You can see the coverage report in CI by looking at the build artifacts in the coverage build in CircleCI).

Reply:
Hi, I'll look at it over the weekend, right now I don't see how you test for exceptions.

Reply:
Hi, I have to say I'm confused by this.
When I put some tracing outputs into the branch and run the coverage tests using `./test/run_envoy_bazel_coverage.sh` I see each of the tests being run twice -- once per IPv4, once per IPv6, the correct branch being taken in each case. Sadly no coverage report as the tests time out on my machine.

Reply:
Yeah, I agree, I think it should be covered. It might be an issue w/ the coverage. Don't worry and thanks for looking!